addons/portal/controllers/portal.py

```diff
@@ -305,22 +305,24 @@ def _check_delivery_address(self, partner_sudo):
         )
         return all(partner_sudo.read(mandatory_delivery_fields)[0].values())
 
-    def _get_mandatory_delivery_address_fields(self, country_sudo):
+    def _get_mandatory_delivery_address_fields(self, country_sudo, **kwargs):
         """ Return the set of mandatory delivery field names.
 
         :param res.country country_sudo: The country to use to build the set of mandatory fields.
         :return: The set of mandatory delivery field names.
         :rtype: set
         """
-        return self._get_mandatory_address_fields(country_sudo)
+        return self._get_mandatory_address_fields(country_sudo, **kwargs)
```

Comment:
I'd return an empty set here instead of calling `_get_mandatory_address_fields` instead.
(that way you don't have the required state_id, street_name and other 'address' fields in localizations).